monitoring/rid_qualifier/flight_track_factory.py

```diff
@@ -0,0 +1,178 @@
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
+                x = point[0]
+                y = point[1]
+                x, y = transformer2.transform(x, y)
+                proj_buffer_points.append((x, y))
+            buffered_path = Polygon(proj_buffer_points)
+            altitude = 70.0
+            flight_points_with_altitude = []
+            x, y = buffered_path.exterior.coords.xy
+            for coord in range(0,len(x)):
+                flight_points_with_altitude.append((x[coord], y[coord], altitude))
+
+            # Build a list of points so that they can be fed to the sim and outputted. 
+            self.flight_points.append(flight_points_with_altitude)
+
+
+class TrackWriter():
+    ''' A class to write data geometry as files in geographical data format''' 
+
+    def __init__(self, path_points, bboxes) -> None:
+
+        self.flight_path_points = path_points
+        self.bboxes = bboxes
+        self.loop_counter = 0
+        
+        output_directory = Path('outputs').mkdir(parents=True, exist_ok=True) # Create outputs directory if it does not exist
+
+    def write_bboxes(self):
+        for buffered_bbox_details in self.bboxes:
+            
+            features = json.dumps({'type': 'Feature', 'properties': {}, 'geometry': shapely.geometry.mapping(buffered_bbox_details['shape'])})
+            with open('outputs/box_%s.geojson'% buffered_bbox_details['name'],'w') as f:
+                f.write(features)
+                
+    def write_tracks(self):
+        ''' This module writes tracks as a GeoJSON / KML for use in other software '''       
+        
+        flight_point_lenghts = {}
+        flight_point_current_index = {}
+        num_flights = len(self.flight_path_points)
+        for i in range(num_flights):
+            flight_point_lenghts[i]= len(self.flight_path_points[i])
+            flight_point_current_index[i] = 0
+            
+        for track_id, flight_track in enumerate(self.flight_path_points):
+            feature_collection = {"type":"FeatureCollection", "features": []}
+            for cur_track_point in flight_track:                
+                p = Point(cur_track_point)
+                point_feature = {'type': 'Feature', 'properties': {}, 'geometry': shapely.geometry.mapping(p)}                
+                feature_collection['features'].append(point_feature)
+
+            with open('outputs/track_%s.geojson'% str(track_id+1),'w') as f:
+                f.write(json.dumps(feature_collection))
+
+            
+
+if __name__ == '__main__':
+    #TODO: accept these parameters as values so that other locations can be supplied
+    my_path_generator = AdjacentCircularFlightPathsGenerator(minx = 7.4735784530639648, miny = 46.9746744128218410, maxx = 7.4786210060119620, maxy= 46.9776318195799121)
+    flight_points = []
+    query_bboxes = []
+    try:
```

Comment:
No need for this `try` block any more; `my_path_generator = AdjacentCircularFlightPathsGenerator(...` will throw an exception if the inputs are invalid.

Reply:
Thanks yes, removing it now.